src/mgr/DaemonServer.cc

```diff
@@ -2261,14 +2261,13 @@ void DaemonServer::adjust_pgs()
 			 << " pg_num " << p.get_pg_num()
 			 << " - still creating initial pgs"
 			 << dendl;
-	      } else if (p.get_pg_num() != p.get_pg_num_pending()) {
+	      } else if (p.get_pg_num() != p.get_pg_num_pending() &&
+			 p.get_pg_num_target() < p.get_pg_num()) {
 		dout(10) << "pool " << i.first
 			 << " target " << p.get_pg_num_target()
 			 << " pg_num " << p.get_pg_num()
-			 << " - pg_num_pending != pg_num, waiting"
+			 << " - decrease and pg_num_pending != pg_num, waiting"
 			 << dendl;
-		// FIXME: we might consider allowing pg_num increases without
-		// waiting for the previously planned merge to complete.
 	      } else if (p.get_pg_num_target() < p.get_pg_num()) {
```

Comment:
I guess the check for permitting a pg merge would then need to be changed to:

```c++
else if (p.get_pg_num() == p.get_pg_num_pending() &&
           p.get_pg_num_target() < p.get_pg_num())
```

Reply:
This is effectively what we currently have on 2275

Reply:
moved the check into the next block to improve readability

Reply:
better!